crypto/evp/names.c

```diff
@@ -94,9 +95,21 @@ const EVP_CIPHER *evp_get_cipherbyname_ex(OSSL_LIB_CTX *libctx,
      */
 
     namemap = ossl_namemap_stored(libctx);
+ retry:
     id = ossl_namemap_name2num(namemap, name);
-    if (id == 0)
-        return NULL;
+    if (id == 0) {
+        EVP_CIPHER *fetched_cipher;
+
+        /* Try to fetch it because the name might not be known yet. */
+        if (!do_retry)
+            return NULL;
+        do_retry = 0;
+        ERR_set_mark();
+        fetched_cipher = EVP_CIPHER_fetch(libctx, name, NULL);
+        EVP_CIPHER_free(fetched_cipher);
+        ERR_pop_to_mark();
+        goto retry;
+    }
```

Comment:
Instead of this retry logic, would it make sense to just do the EVP_CIPHER_fetch, and return that value if its not null?  If it is NULL, then walk through the OBJ_to_name translation?  i.e.:

```
const EVP_CIPHER *evp_get_cipherbyname_ex(OSSL_LIB_CTX *libctx,
                                          const char *name)
{
     EVP_CIPHER *cp = EVP_CIPHER_fetch(libctx, name, NULL);
     if (cp != NULL)
          return cp;
     cp = (const EVP_CIPHER *)OBJ_NAME_get(name, OBJ_NAME_TYPE_CIPHER_METH);
     ....
```

That seems like it would be more compact and performant

Reply:
you cant do that because this is a dynamic pointer that needs to be freed then - the one it does return is constant